packages/cli/src/models/network/ProjectDeployer.ts

```diff
@@ -61,7 +61,7 @@ export class SimpleProjectDeployer extends BaseProjectDeployer {
     this.project = new SimpleProject(this.packageFile.name, this.txParams);
     this.networkFile.version = this.requestedVersion;
     _.forEach(this.networkFile.contracts, (contractInfo, contractAlias) => {
-      this.project.registerImplementation(contractAlias, contractInfo);
+      this.project.registerImplementation(contractAlias, { address: contractInfo.address, bytecodeHash: contractInfo.bodyBytecodeHash });
```

Comment:
Great! I've modified it to use "localBytecodeHash" instead of "bodyBytecodeHash". No hurry on this, let's wait for Jota's review.

Reply:
Mmmh... It's clear that having three different `bytecodehash`es is somewhat annoying :disappointed:. Let me clarify you each of them so we can decide which one should we use:
- `localBytecodeHash`: it's the hashed version of the solidity bytecode (constructor + body code) with solidity libraries' placeholders unaltered.
- `deployedBytecodeHash`: It's the hashed version of the solidity bytecode (constructor + body code) with solidity libraries' placeholders replaced with the corresponding addresses.
- `bodyBytecodeHash`: It's the `deployedBytecodeHash` without the constructor code. 

In other words, the main difference between `localBytecodeHash` and `deployedBytecodeHash` is that the former has the solidity libs placeholders unaltered, and the latter has the solidity libs placeholders replaced by its addresses. In addition to this, `bodyBytecodeHash` is the `deployedBytecodeHash` without the constructor code.

That being explained... Shouldn't we be using `deployedBytecodeHash` instead of `localBytecodeHash`, and replace all the `contractClass.bytecode`with `contractClass.deployedBytecode` (for instance, in `lib/src/project/SimpleProject.ts#L78`)? @spalladino @ajsantander

Reply:
Looking at the code for a bit, I see that SimpleProject's `this.implementations: { [contractName: string]: { address: string, bytecodeHash } }` is where the "bytecodeHash" is stored, but I can't find any place where that "bytecodeHash" is ever required for use... 

Having said that, and given that these are implementations, I believe that constructors shouldn't be relevant because these contracts are meant to be proxied, right? In that case, constructors are irrelevant, and we'd only want "bodyBytecodeHash".

Reply:
> Looking at the code for a bit, I see that SimpleProject's `this.implementations: { [contractName: string]: { address: string, bytecodeHash } }` is where the "bytecodeHash" is stored, but I can't find any place where that "bytecodeHash" is ever required for use...

I believe we do use the `bytecodeHash` in `SimpleProject#_getOrDeployOwnImplementation` to check if the implementation contract exists (and hasn't changed) before trying to deploy it.  OTOH, I agree that, as these contracts are meant to be proxied, we should use `bodyBytecodeHash`. Would you mind if I commit a comment to `Bytecode.ts`'s `splitCode` function explaining each `xxxxBytecodeHash` in this PR?

Reply:
Of course not! Would you be a cutie and also change back SimpleProjectDeployer to use bodyBytecodeHash instead of localBytecodeHash?